diff --git dlls/d3d8/tests/visual.c dlls/d3d8/tests/visual.c
index f1e648a..8387c34 100644
--- dlls/d3d8/tests/visual.c
+++ dlls/d3d8/tests/visual.c
@@ -12565,6 +12565,7 @@ START_TEST(visual)
         return;
     }
 
+#ifndef WINETEST_NO_WOW64
     if (!is_win64 && IsWow64Process( GetCurrentProcess(), &is_wow64 ) && is_wow64)
     {
         TEB64 *teb64 = ULongToPtr( NtCurrentTeb()->GdiBatchCount );
@@ -12574,6 +12575,7 @@ START_TEST(visual)
             new_wow64 = !!peb64->LdrData;
         }
     }
+#endif
 
     memset(&identifier, 0, sizeof(identifier));
     hr = IDirect3D8_GetAdapterIdentifier(d3d, D3DADAPTER_DEFAULT, 0, &identifier);
